whedon.rb

```diff
@@ -211,3 +272,38 @@ def check_editor
     halt 403
   end
 end
+
+class WhedonWorker
+  include Sidekiq::Worker
+
+  def perform(papers, site_host, site_name, nwo, issue_id)
+    bg_respond("Hello from the background worker", nwo, issue_id)
+    set_env(papers, site_host, site_name, nwo)
+    download(issue_id)
+    compile(issue_id)
+  end
+
+  def download(issue_id)
+    puts "Downloading #{ENV['REVIEW_REPOSITORY']}"
+    `whedon download #{issue_id}`
+  end
+
+  def compile(issue_id)
+    puts "Compiling #{ENV['REVIEW_REPOSITORY']}/#{issue_id}"
+    `whedon prepare #{issue_id}`
+  end
+
+  def bg_respond(one, two, three)
+    puts ENV
+  end
+
+  # The Whedon gem expects a bunch of environment variables to be available
+  # and this method sets them.
+  def set_env(papers, site_host, site_name, nwo)
```

Comment:
This is kind of weird. We have to basically set all of these environment variables as the Whedon gem needs them to compile the paper.